apps/civicsignalblog/src/payload/access/canAccessApplication.js

```diff
@@ -1,7 +1,11 @@
+import { ROLE_APP } from "./roles";
+
 export default function canAccessApplication(user, searchString) {
+  // We are using this condition to only control what a user can see at a given time
   if (user) {
     const app = user.currentApp || user.defaultApp;
-    return app === searchString.toLowerCase();
+    // The assumption is that one account with ROLE_APP will be used to access all the app content
+    return app === searchString.toLowerCase() || user.roles.includes(ROLE_APP);
```

Comment:
I was thinking in the long run we would want the `ROLE_APP` to only have full read access, but no update/create/delete privileges

Reply:
Yes, we could introduce this role; however, the limitation with the current implementation is that `canAccessApplication` only restricts what the user can view based on the application they have selected—specifically by using the `currentApp` field in their profile.  i.e. each user we create has access to all apps, `canAccessApplication` controls only what collections they can see based on the app they have selected.

We could change the `currentApp` to support the selection of multiple apps so that we can accommodate what the subscriber (API) can access.

Reply:
I'm worried that this approach wouldn't work well if we want one Payload user to access content across all the apps, I'm not sure how well payload would handle concurrent requests from different apps (my concern is that this could result into one app being denied access to the content)...I think its best if we limit  the number of times we update the `currentApp` field  i.e. just do it on the backend during account creation

Reply:
Not a bad thing to worry about but if it is that _fragile_ then the `currentApp` approach is probably not the best option for what we need.

Essentially, we should be able to make a request: I want this data from this app. These requests should be self contained i.e. one requests is not affected by the previous/next.

Authentication should be able to tell us yes/no, this use is allowed/not allowed to read/write this data to this part of the app.

Reply:
I think the best approach would be to Add `allowed apps` field that will track what apps users can access.

We will distinguish API and normal backend UI permission requests with the header we'll be sending. i.e. With app requests we will only check if the user can access the app they want content from, with backend UI we will put an additional check on the `currentApp` selected by the user. 

This way we wont need to make any updates on the `currentApp` per each request we'll be sending via the API.

Reply:
... it's a solid approach but isn't this (the `currentApp` or `app`, etc.) the kind of thing that should always be submitted per request?

1. APIs: Header, etc.
2. Browser/Backend UI: (session/secure) cookie. You set it in the browser and the browser submit it with every request.

I'm just hesitant on approaches that require two different ways of doing basically the same thing.

Reply:
Cool Let me explore the best way we can set session values in Payload for the browser/backend UI, I think there should be a clean way to do it...this would remove the `currentApp` field on the users collection and require to move a few parts on the app selection component

Reply:
@kilemensi, Can you have another look, I decided to keep the `currentApp` field, as I couldn't safely set session variables for the browser/backend UI, the next best thing was to try out [user preferences](https://payloadcms.com/docs/admin/preferences) but getting the values from the access control functions would need us to query the database (which adds more complexity, while as opposed to using the `user` object which is automatically provided from `req`), 

I think with these modifications we can have a more consistent flow for permission checks and would be the same regardless of where we are getting the `currentApp` from.

I have made some assumptions on the permissions for the collections i.e. _ROLE_EDITOR_, _ROLE_ADMIN_ are the only roles that can _update_, _create_, _delete_ collections as opposed to having them open to any user.